oioioi/sinolpack/utils.py

```diff
@@ -1,7 +1,42 @@
+import logging
+
 from oioioi.filetracker.utils import django_to_filetracker_path
 from oioioi.sinolpack.models import ExtraConfig, ExtraFile
+from oioioi.evalmgr import tasks
+
+logger = logging.getLogger(__name__)
 
 
+def add_extra_execution_files(environ):
+    try:
+        config = ExtraConfig.objects.get(problem_id=environ.get('problem_id')) \
+                 .parsed_config
+    except ExtraConfig.DoesNotExist:
+        config = {}
+
+    lang = environ.get('language')
+    for _, job in environ.get('tests', {}).items():
+        extra_args = config.get('extra_execution_args', {}).get(lang, [])
+        if isinstance(extra_args, str):
+            extra_args = [extra_args]
+        if extra_args:
```

Comment:
if extra_args is `""` initially, then it will be transformed to `[""]`  which is truthy so it will be propagated.
It's an edge case, and I don't even know if it will break anything downstream, and it looks like
```
extra_execution_args:
  c:
```
results in None, not "" - you'd have to explicitly write
```
extra_execution_args:
  c: ""
```
which someone could conceivably write with the intention of adding an empty argument, so I guess we can leave it as is, though in an ideal world I'd want to see a comment in the code explaining why this is ok.

Reply:
I see the point. It could use some touch ups and reformatting for sure.